src/axom/primal/operators/detail/evaluate_integral_impl.hpp

```diff
@@ -0,0 +1,197 @@
+    {
+      dCarray[p] = c[p][i];
+    }
+
+    // stop one step early and take difference of last two values
+    for(int p = 1; p <= ord - 1; ++p)
+    {
+      const int end = ord - p;
+      for(int k = 0; k <= end; ++k)
+      {
+        dCarray[k] = (1 - t) * dCarray[k] + t * dCarray[k + 1];
+      }
+    }
+    val[i] = ord * (dCarray[1] - dCarray[0]);
+  }
+
+  return val;
+}
+
+/*!
+ * \brief Evaluate a scalar field line integral on a single Bezier curve.
+ *
+ * Evaluate the scalar field line integral with MFEM integration rule
+ *
+ * \param [in] c the Bezier curve object
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input and return a double
+ * \param [in] quad the mfem integration rule containing nodes and weights
+ * \return the value of the integral
+ */
+inline double evaluate_line_integral_component(
+  const primal::BezierCurve<double, 2>& c,
+  std::function<double(Point2D)> integrand,
+  const mfem::IntegrationRule* quad)
+{
+  // Store intermediate values
+  Point2D x_q;
+  Vector2D dx_q;
+
+  // Store/compute quadrature result
+  double full_quadrature = 0.0;
+  for(int q = 0; q < quad->GetNPoints(); q++)
+  {
+    // Get intermediate quadrature point
+    //  at which to evaluate tangent vector
+    x_q = c.evaluate(quad->IntPoint(q).x);
+    dx_q = detail::temp_dt(c, quad->IntPoint(q).x);
+
+    full_quadrature += quad->IntPoint(q).weight * integrand(x_q) * dx_q.norm();
+  }
+
+  return full_quadrature;
+}
+
+/*!
+ * \brief Evaluate a vector field line integral on a single Bezier curve.
+ *
+ * Evaluate the vector field line integral with MFEM integration rule
+ *
+ * \param [in] c the Bezier curve object
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input and return a 2D axom vector object
+ * \param [in] quad the mfem integration rule containing nodes and weights
+ * \return the value of the integral
+ */
+inline double evaluate_line_integral_component(
+  const primal::BezierCurve<double, 2>& c,
+  std::function<Vector2D(Point2D)> vec_field,
+  const mfem::IntegrationRule* quad)
+{
+  // Store intermediate values.
+  Point2D x_q;
+  Vector2D dx_q, func_val;
+
+  // Store/compute quadrature result
+  double full_quadrature = 0.0;
+  for(int q = 0; q < quad->GetNPoints(); q++)
+  {
+    // Get intermediate quadrature point
+    //  on which to evaluate dot product
+    x_q = c.evaluate(quad->IntPoint(q).x);
+    dx_q = detail::temp_dt(c, quad->IntPoint(q).x);
+    func_val = vec_field(x_q);
+
+    full_quadrature +=
+      quad->IntPoint(q).weight * Vector2D::dot_product(func_val, dx_q);
+  }
+
+  return full_quadrature;
+}
+
+/*!
+ * \brief Evaluate the integral across a single component of the curved polygon.
+ *
+ * Uses a Spectral Mesh-Free Quadrature derived from Green's theorem, evaluating
+ * the area integral as a line integral of the antiderivative over the curve.
```

Comment:
Please clarify that this function is meant to be called for each `BezierCurve` of a closed polygon, and add a (simplified) reference to the paper